src/VAC/View.cpp

```diff
@@ -1087,46 +1093,221 @@ void View::PMRReleaseEvent(int action, double x, double y)
+        vac_->continueSketchEdge(getX((180 + 360 * 3) / 7), getY((180 + 360 * 3) / 7), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX((180 + 360 * 3) / 7), getY((180 + 360 * 3) / 7), w, interactiveTime());
+        vac_->continueSketchEdge(getX((180 + 360 * 4) / 7), getY((180 + 360 * 4) / 7), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX((180 + 360 * 4) / 7), getY((180 + 360 * 4) / 7), w, interactiveTime());
+        vac_->continueSketchEdge(getX((180 + 360 * 5) / 7), getY((180 + 360 * 5) / 7), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX((180 + 360 * 5) / 7), getY((180 + 360 * 5) / 7), w, interactiveTime());
+        vac_->continueSketchEdge(getX((180 + 360 * 6) / 7), getY((180 + 360 * 6) / 7), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX((180 + 360 * 6) / 7), getY((180 + 360 * 6) / 7), w, interactiveTime());
+        vac_->continueSketchEdge(getX((180 + 360 * 7) / 7), getY((180 + 360 * 7) / 7), w);
+        vac_->endSketchEdge();
+
+        emit allViewsNeedToUpdatePicking();
+        updateHoveredObject(mouse_Event_X_, mouse_Event_Y_);
+        emit allViewsNeedToUpdate();
+    }
+    else if(action == OCTAGON_ACTION)
+    {
+        vac_->endRectangleOfSelection();
+        lastMousePos_ = QPoint(mouse_Event_X_,mouse_Event_Y_);
+
+        vac_->beginSketchEdge(getX(0), getY(0), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45), getY(45), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45), getY(45), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 2), getY(45 * 2), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 2), getY(45 * 2), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 3), getY(45 * 3), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 3), getY(45 * 3), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 4), getY(45 * 4), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 4), getY(45 * 4), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 5), getY(45 * 5), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 5), getY(45 * 5), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 6), getY(45 * 6), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 6), getY(45 * 6), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 7), getY(45 * 7), w);
+        vac_->endSketchEdge();
+
+        vac_->beginSketchEdge(getX(45 * 7), getY(45 * 7), w, interactiveTime());
+        vac_->continueSketchEdge(getX(45 * 8), getY(45 * 8), w);
+        vac_->endSketchEdge();
+
+        emit allViewsNeedToUpdatePicking();
+        updateHoveredObject(mouse_Event_X_, mouse_Event_Y_);
+        emit allViewsNeedToUpdate();
```

Comment:
These three lines are in all cases. It would be more readable with a function IMO

Reply:
Done